### DISCUSSION TITLE
A new service to set "charge to add" on Intelligent Octopus Go for EV charging

### DISCUSSION THREAD

**Discussion by stevedundee2** [BottlecapDave/HomeAssistant-OctopusEnergy#801] [Ideas]
It would be nice to have a new service to set "charge to add" on Intelligent Octopus Go for EV charging.

Octopus can control some EV charges, e.g. Zappi. You have to tell Octopus through their app how many percentage points of charge you need to add. In the app this is called "charge to add". So if are down to 20% and want 80%, then you tell them 60%. But there is no service for this (and perhaps nothing in the Octopus API in the first place - I don't know how to find out).

My use case is a an automation triggered by plugging in the car: get the target and current charge percentages from the car (I can get these from the VW integration), work out the difference, increase if necessary to whole 5% boundary (e.g. currently 19%, want 80%, so need to set "charge to add" to 65%), and use that to set "charge to add".

Just a nice to have - save me from accessing the Octopus app to do it.

**Comment by BottlecapDave** [comment]
Hello and sorry for the late response.

I _believe_ this is covered by the [charge limit entity](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/entities/intelligent/#charge-limit) where you can set the amount you desire your car to be at by the specified time.

**Reply by stevedundee2** [reply]
Ah yes sorry. I thought that was read only. You are right you can set it and it works perfectly. Thanks for replying to such a dumb fool question.

Steve
